server.js

```diff
@@ -23,58 +24,53 @@ app.use("/public", express.static(__dirname + "/public"));
-            res.render('index');
-            return;
-        }
-        //try to get the current session from DB. If there's not, clear the useless cookie.
-        const currentSession = await Session.findById(req.cookies.session);
-        if (!currentSession) {
-            res.clearCookie('session');
-            res.render('index');
-            return;
-        }
-        //try to find the user based on the session. If not found, clear the useless cookie
-        const currentUser = await User.findById(currentSession.user);
-        if (!currentUser) {
-            res.clearCookie('session');
-            res.render('index');
-            return
-        }
-        // If it got here, it means that the user is authenticated and ready to go to personal profile page.
-        res.redirect('/profile');
-    } catch (err) {
-        res.render('error', { "error message": err })
+app.use(async (req, res, next) => {
+    if (!req.cookies.session) {
+        next()
+        return;
+    }
+    const currentSession = await Session.findById(req.cookies.session);
+    if (!currentSession) {
+        res.clearCookie('session');
+        next()
+        return;
+    }
+    const currentUser = await User.findById(currentSession.user);
+    if (!currentUser) {
+        res.clearCookie('session');
+        next()
+        return;
     }
+    req.session = currentSession
+    req.user = currentUser
+    console.log(req.user);
+    next()
 })
 
-app.get("/profile", async (req, res) => {
+app.use('/profile', profile);
+app.use('/post', post);
+app.use('/login', login);
+app.use('/signup', signup);
+
+// get requests
+app.get("/", async (req, res) => {
     try {
-        //check if there's a session. If not, direct to the loging main page. In case someone try to get to profile page without authentication.
-        if (!req.cookies.session) {
-            res.redirect('/')
-            return;
-        }
-        // check if session is in the database. If not, direct to main page. (the / will clear the cookie)
-        const currentSession = await Session.findById(req.cookies.session);
-        if (!currentSession) {
-            res.redirect('/')
-            return;
-        }
-        // find user base on the session
-        const currentUser = await User.findById(currentSession.user);
-        if (!currentUser) {
-            res.redirect('/')
+        const findPosts = await Post.find({})
+        const posts = findPosts.map(item => {
+            return {
+                id: item.id,
+                title: item.title,
+                time: item.created,
+                user: item.user.name
+            }
+        }).reverse()
+        if (!req.user) {
+            res.render('forum', { posts: posts });
             return;
         }
-        // if it got here it means the user is authenticated.
-        res.render("profile", currentUser)
+        res.render('forum', { user: req.user, posts: posts, admin: req.user.admin });
```

Comment:
```suggestion
        res.render('forum', { user: req.user, posts: posts, admin: req.user?.admin });
```

If there is no `req.user`, you'd just pass `undefined` for `user` which is perfect for the template. Same with `admin`. The `?.` operator means "go in only if exists, if not return `undefined`" So if `req.user` is `undefined`